### PR TITLE
Use find_package for llvm/clang

### LINKED ISSUE #7176: Unable to build with external LLVM 9
When testing #6385 I've had issues with the linking order of the LLVM libs due to CMake not knowing about their dependencies. I don't have a log or reproducer to hand but this patch was enough to fix it:

https://github.com/chrisburr/root-feedstock/blob/4dff10152d66bd0e873b60150e72cf0f9bfa8c19/recipe/patches/0001-Use-find_package-for-llvm-clang.patch

The `get_directory_property` call was also an issue. That patch contains the values copied from a build with `builtin_llvm=ON`.

### PR DESCRIPTION
**Pull request by chrisburr**
Fixes #7176

### CODE REVIEW

File: interpreter/CMakeLists.txt [hunk 1]
```diff
@@ -449,7 +452,7 @@ if (builtin_cling)
   # LLVM doesn't really give us a API to get this with an in-source build
   # so we just use the normal way of doing this and read the llvm directory
   # compilation properties.
-  get_directory_property(LLVM_DEFS DIRECTORY llvm/src COMPILE_DEFINITIONS)
+  set(LLVM_DEFS "_GNU_SOURCE;__STDC_CONSTANT_MACROS;__STDC_FORMAT_MACROS;__STDC_LIMIT_MACROS")
```
Comment (vgvassilev) [inline interpreter/CMakeLists.txt:455]: `_GNU_SOURCE` seems to be defined only if `LLVM_USING_GLIBC` is set. That's probably what breaks osx.

Comment (chrisburr) [reply interpreter/CMakeLists.txt:455]:
```suggestion
  if( LLVM_USING_GLIBC )
    set(LLVM_DEFS "_GNU_SOURCE;__STDC_CONSTANT_MACROS;__STDC_FORMAT_MACROS;__STDC_LIMIT_MACROS")
  endif()
```

File: interpreter/CMakeLists.txt [hunk 2]
```diff
@@ -449,7 +452,9 @@ if (builtin_cling)
   # LLVM doesn't really give us a API to get this with an in-source build
   # so we just use the normal way of doing this and read the llvm directory
   # compilation properties.
-  get_directory_property(LLVM_DEFS DIRECTORY llvm/src COMPILE_DEFINITIONS)
+  if( LLVM_USING_GLIBC )
+    set(LLVM_DEFS "_GNU_SOURCE;__STDC_CONSTANT_MACROS;__STDC_FORMAT_MACROS;__STDC_LIMIT_MACROS")
+  endif()
```
Comment (vgvassilev) [inline interpreter/CMakeLists.txt:457]:
```suggestion
set(LLVM_DEFS "__STDC_CONSTANT_MACROS;__STDC_FORMAT_MACROS;__STDC_LIMIT_MACROS")
  if( LLVM_USING_GLIBC )
    list(APPEND LLVM_DEFS "_GNU_SOURCE")
  endif()
```

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Review by vgvassilev — APPROVED** [review]
LGTM!

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by vgvassilev** [comment]
@chrisburr, apologies for the delay. Is this PR still relevant? If so could you rebase it?

**Comment by guitargeek** [comment]
> @chrisburr, apologies for the delay. Is this PR still relevant? If so could you rebase it?

Since the author of the PR didn't follow up on this request from one and a half years ago, I'll close this PR as stalled.

But of course, should this be relevant for you again, feel free to re-open it or create a new PR!

**Pull request closed without merging**